microprofile/tests/junit5/src/main/java/io/helidon/microprofile/tests/junit5/HelidonJunitExtension.java

```diff
@@ -163,6 +169,35 @@ public void beforeAll(ExtensionContext context) {
         return (T[]) result;
     }
 
+    private AddConfig[] scanBeansConfig() {
+        if (!testClass.isAnnotationPresent(ScanBeansForConfig.class)) return new AddConfig[0];
+        List<AddConfig> result = new ArrayList<>();
+        for (AddBean addBean : classLevelBeans) {
+            Class<?> beanClazz = addBean.value();
+            result.addAll(List.of(getAnnotations(beanClazz, AddConfig.class)));
+        }
+        return result.toArray(new AddConfig[0]);
+    }
+
+    private void beforeContainer() {
+        Class<?> clazz = testClass;
+        try {
+            while (clazz != Object.class) {
+                for (Method m : clazz.getDeclaredMethods()) {
+                    if (Objects.isNull(m.getAnnotation(BeforeContainer.class))) continue;
+                    if (!Modifier.isStatic(m.getModifiers())) {
+                        throw new RuntimeException("Method annotated with @BeforeContainer has to be static");
+                    }
+                    m.setAccessible(true);
+                    m.invoke(null);
+                }
+                clazz = clazz.getSuperclass();
+            }
+        } catch (IllegalAccessException | InvocationTargetException e) {
+            throw new RuntimeException("Cannot invoke beforeContainer method", e);
+        }
+    }
+
```

Comment:
@tomas-langer FYI needed a way to run test MQTT broker before CDI is started, also ability to use `@AddConfig` on test beans simplify my tests a lot by avoiding [uber test configs](https://github.com/oracle/helidon/blob/master/tests/integration/jms/src/test/java/io/helidon/messaging/connectors/jms/JmsMpTest.java#L64-L143)